src/collections/Breadcrumb/Breadcrumb.js

```diff
@@ -25,28 +25,23 @@ function Breadcrumb(props) {
     return <div className={classes} {...rest}>{children}</div>
   }
 
-  let dividerJSX = <Breadcrumb.Divider />
+  const dividerJSX = <Breadcrumb.Divider icon={icon}>{divider}</Breadcrumb.Divider>
+  const sectionsJSX = []
 
-  if (divider) {
-    dividerJSX = <Breadcrumb.Divider>{divider}</Breadcrumb.Divider>
-  }
+  sections.forEach(({ text, ...restSection }, index) => {
+    const key = `${text}-${index}`
+    const dividerKey = `$divider-${index}`
 
-  if (icon) {
-    dividerJSX = <Breadcrumb.Divider icon={icon} />
-  }
+    sectionsJSX.push(
+      <Breadcrumb.Section {...restSection} key={key}>{text}</Breadcrumb.Section>
+    )
 
-  return (
-    <div className={classes} {...rest}>
-      {sections.map((section, index) => {
-        const isLast = index === (sections.length - 1)
-
-        return [
-          <Breadcrumb.Section {...section}>{section.text}</Breadcrumb.Section>,
-          isLast ? null : dividerJSX,
-        ]
-      })}
-    </div>
-  )
+    if (index !== sections.length - 1) {
+      sectionsJSX.push(React.cloneElement(dividerJSX, { key: dividerKey }))
```

Comment:
Whoops!  You're right, I reused the same `key` on the divider as I did on the section.  React uses the `key` to identify components, so it thought this was one component.

There may be a typo though:

```js
const dividerKey = `$divider-${index}`
//=> `$divider-0`, `$divider-1`, etc.
```

You may have intended to do this?

```js
const dividerKey = `${divider}-${index}`
//=> `/-0`, `/-1`, etc.
```

I would suggest we just add `divider` to the `key` for the `dividerKey`. Given a breadcrumb like so: `Home / Music / Playlist`, we'd have:

```js
const key = `${text}-${index}`
const dividerKey = `${key}-divider`
//=> key: `Home-0`, `Music-1`, `Playlist-1`
//=> dividerKey: `Home-0-divider`, `Music-1-divider`, `Playlist-1-divider`
```

This should be reasonably unique for now.  Each element just needs to have a unique `key`.  Open to better ideas.